RFC-0010-jdk-upgrade.md

```diff
@@ -0,0 +1,293 @@
+# **RFC 10 for Presto**
+
+## JDK Upgrade for Presto
+
+Proposers
+
+* ZacBlanco
+* imjalpreet
+* aaneja
+
+
+## Summary
+
+This RFC lists out the steps and process for upgrading the build and runtime JDK version used in Presto to a more recent LTS release like 17 and 21  
+As of today (2025-01-24) the version we use is JDK 8
+
+
+## Background
+
+The Java community has slowly but surely been moving to JDK versions beyond JDK 8  
+
+Some of our dependencies, e.g [Jetty](https://github.com/jetty/jetty.project/) have already moved to JDK 17, and declared prior versions to be out-of-support(See https://github.com/jetty/jetty.project/issues/10485. The latest supported version is Jetty 12)
+
+Not using modern versions of our dependencies opens Presto installations to security risks through CVEs, which hurts adoption of Presto
+
+See this [Prestocon 2024 presentation](https://www.youtube.com/watch?v=K6CjuPZtqW4&list=PLJVeO1NMmyqW_qoMMEyVq-wW8lBSxA7hP&index=10) for a deep-dive motivating the issue further (Slides [here TODO]())
+
+
+## Proposed Migration Plan  
+
+To migrate to newer versions of the JDK, we propose a phased approach -   
+
+![slide from Prestocon 2024 presentation](RFC-0010/jdk_upgrade_timeline.png)
```

Comment:
Since we're trying to be generic in this RFC I feel like it would be better to just include a table in text, and remove mention of the specific versions

| Phase | Time Window | Release Version | Runtime JDK Support | Compilation JDK Support |
|--------|--------|--------|--------|--------|
| Dual Support | Month N | v0.X | JDK Y and Y+1 | JDK Y |
| Dual Support & Deprecation | Month N + 2 | v0.(X+1) | JDK Y and Y+1 | JDK Y |
| Stable JDK | Month N + 4 | v0.(X+2) | JDK Y+1 | JDK Y+1 |

Reply:
Cool, thanks for building the table!